device/src/storage.rs

```diff
@@ -1,73 +1,122 @@
 extern crate alloc;
-use crate::state::FrostState;
-use alloc::format;
+use alloc::{format, string::String};
 use bincode::{
     de::read::Reader,
     enc::write::Writer,
     error::{DecodeError, EncodeError},
 };
 use embedded_storage::{ReadStorage, Storage};
 use esp_storage::{FlashStorage, FlashStorageError};
+use frostsnap_core::{message::DeviceToStorageMessage, schnorr_fun::fun::Scalar};
 
-pub const NVS_PARTITION_START: u32 = 0x9000;
-pub const NVS_PARTITION_SIZE: usize = 0x6000;
+const NVS_PARTITION_START: u32 = 0x9000;
+const _NVS_PARTITION_SIZE: usize = 0x6000;
+const HEADER_LEN: usize = 256;
+const DATA_START: u32 = NVS_PARTITION_START + HEADER_LEN as u32;
+const MAGIC_BYTES_LEN: usize = 8;
+const MAGIC_BYTES: [u8; MAGIC_BYTES_LEN] = *b"fsheader";
 
-pub struct DeviceStorageRw<'a> {
-    nvs: &'a mut DeviceStorage,
+pub struct DeviceStorage {
+    flash: FlashStorage,
     pos: u32,
 }
 
-pub struct DeviceStorage {
-    flash: FlashStorage,
-    start_pos: u32,
+#[derive(Debug, Clone, bincode::Encode, bincode::Decode)]
+pub enum Change {
+    Core(DeviceToStorageMessage),
+    Name(String),
+}
+
+#[derive(Debug, Clone, bincode::Encode, bincode::Decode)]
+pub struct Header {
```

Comment:
I want to understand the Header better: is this for data which will never change? What else might end up going in here?

Reply:
Yes. Never change unless it's factory reset.

Reply:
Maybe silly question, but in this case do we need a header at all? Or maybe in future we'll purge some of change set?